internal/api/token.go

```diff
@@ -333,15 +333,26 @@ func (a *API) generateAccessToken(r *http.Request, tx *storage.Connection, user
 			ExpiresAt: jwt.NewNumericDate(expiresAt),
 			Issuer:    config.JWT.Issuer,
 		},
-		Email:                         user.GetEmail(),
-		Phone:                         user.GetPhone(),
-		AppMetaData:                   user.AppMetaData,
-		UserMetaData:                  user.UserMetaData,
-		Role:                          user.Role,
-		SessionId:                     sid,
-		AuthenticatorAssuranceLevel:   aal.String(),
-		AuthenticationMethodReference: amr,
-		IsAnonymous:                   user.IsAnonymous,
+		AuthenticatorAssuranceLevel: aal.String(),
+		SessionId:                   sid,
+		Role:                        user.Role,
+		IsAnonymous:                 user.IsAnonymous,
```

Comment:
I think it's alright for `is_anonymous` and `aal` to not be required claims since they are only relevant for folks who:
1. Have anonymous sign-ins enabled
2. Use MFA (need to write RLS policies to check against the AAL)

I didn't test this but i'm guessing it's required because if those claims are omitted, they may break existing functionality if either (1) or (2) are enabled?

Reply:
yup, I initially made them optional but decided to wait for the decision out of #1913 if we should have them. Some tests also started breaking with these omitted: https://github.com/supabase/auth/actions/runs/12866198657/job/35868267108#step:11:1147, which further motivated me to keep them as it seem a good indicator that they might be required

Reply:
thought about this a bit more and i think it's safe to let the user decide whether `is_anonymous` and `aal` should be present in the JWT as long as we make it clear in the configuration UI that existing RLS policies that check against these claims would fail